logbook-netty/src/main/java/org/zalando/logbook/netty/Request.java

```diff
@@ -56,19 +64,29 @@ public String getMethod() {
 
     @Override
     public String getScheme() {
-        // TODO pick the real one
-        return "http";
+        SslHandler handler = context.channel().pipeline().get(SslHandler.class);
+        return handler == null ? "http" : "https";
     }
 
     @Override
     public String getHost() {
-        return request.headers().get(HOST, "unknown");
+        String host = request.headers().get(HOST);
+        if (host != null) {
+            int separator = host.indexOf(":");
+            return separator == -1 ? host : host.substring(0, separator);
```

Comment:
Would be easier to understand if it were a small private method with a descriptive name, e.g. `stripPortIfNecessary`.

Reply:
done